### DISCUSSION TITLE
Audio out locally?

### DISCUSSION THREAD

**Discussion by maddin06** [Thyraz/Sonos-Kids-Controller#4] [General]
Hi, 

As you initially wrote, there are several drawbacks when using a RFID driven box. Especially when it comes to "maintain the cards"... 

Currently, I'm using [this project here](https://github.com/biologist79/Tonuino-ESP32-I2S) with an ESP32 and the webgui provided by the project to fight against the card chaos. Good thing is, everything is local. There is no need to run it in my home network. But I'm missing Spotify somehow to have endless audio/music available. The kids are getting older.

Meanwhile, as I have more listeners at home, I'm playing around with mpd, mopidy and volumio. I would like to have it with a simple touchscreen interface a 4 or 8 year old kid is able to understand. Volumios problem is the slow touchscreen interface with large libraries. For mpd and mopidy I was not able to find anything child-friendly (or I'm blind). 

I found your project already some weeks ago and I like the interface and spotify integration. There is a Sonos in my network, so your project might fit. But I only have one Sonos and I won't buy a Sonos for every kid here at home ;) 

**TL;DR**
I haven't reviewed your code yet as I'm not very familiar with "web tech". Just from your feeling and insight, how hard could it be to use a local output like the headphone jack or e.g. a hifiberry amp? Or, the other way around, what about a port to mopidy for the interface? I really like it.

**Comment by Thyraz** [comment]
Hi maddin,

this software is just a graphical remote for this node sonos http library: https://github.com/jishi/node-sonos-http-api/tree/master/lib/actions

So this isn't possible currently.

But the player controls are just a few http requests.
If you have a library or player software to locally play music e.g. from Spotify that can be controlled through http requests,
you might be able to adjust this file for your needs:

https://github.com/Thyraz/Sonos-Kids-Controller/blob/master/src/app/player.service.ts

Regards,
Tobias
